src/App.jsx

```diff
@@ -14,33 +17,86 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reversed, setReversed] = useState(false);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const getSortedGoods = () => {
+    const sortedGoods = [...goodsFromServer].sort((good1, good2) => {
+      switch (sortField) {
+        case 'alphabetically':
+          return good1.localeCompare(good2);
+        case 'length':
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    if (reversed) {
+      sortedGoods.reverse();
+    }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    return sortedGoods;
+  };
+
+  const visibleGoods = getSortedGoods();
```

Comment:
The `visibleGoods` variable is calculated on every render. Consider using `useMemo` to memoize the sorted goods, which can improve performance by avoiding unnecessary recalculations when dependencies haven't changed.